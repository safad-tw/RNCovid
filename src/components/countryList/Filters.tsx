import * as React from 'react';
import { useState } from 'react';
import {
  Text,
  View,
  TouchableOpacity,
  StyleSheet
} from 'react-native';


interface FilterProps {
  data: any,
  onValueChange: (data: number) => void
}

const Filters = ({ data, onValueChange }: FilterProps) => {
  const [selectedIndex, setSelectedIndex] = useState(-1);

  return (
    <View style={{ flexDirection: 'row' }}>
      {data.map((x: any, i: number) => (
        <FilterButton
          text={x.label}
          id={i}
          key={i}
          selectedIndex={selectedIndex}
          callback={(id) => {
            setSelectedIndex(id);
            if (onValueChange) {
              onValueChange(id);
            }
          }}
        />
      ))}
    </View>
  );
};

export interface FilterButtonProps {
  callback: (data: number) => void, id: number, selectedIndex: number, text: string
}

const FilterButton = ({ callback, text, id, selectedIndex }: FilterButtonProps) => {
  const clicked = selectedIndex === id;
  return (
    <TouchableOpacity
      style={[
        styles.buttonStyle,
        { backgroundColor: clicked ? 'black' : 'white' },
      ]}
      onPress={() => {
        callback(id);
      }}>
      <Text style={{ color: clicked ? 'white' : 'black' }}>
        {text}
      </Text>
    </TouchableOpacity>
  );
};


const styles = StyleSheet.create({
  buttonStyle: {
    borderRadius: 20,
    borderColor: 'black', 
    borderWidth: 2, 
    padding: 10, 
    margin: 5
  }
})

export default Filters;