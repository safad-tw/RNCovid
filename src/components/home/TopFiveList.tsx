import * as React from 'react';
import {
    Text,
    View,
    StyleSheet,
    Dimensions, FlatList, Button
} from 'react-native';

import LanguagesStrings from './../../localization/LanguagesStrings';
import {Country} from './../../models/Country'

interface TopFiveListProps {
    data: [Country],
    navigateTo: () => void;
}

class TopFiveList extends React.Component<TopFiveListProps> {

    constructor(props: TopFiveListProps) {
        super(props);
    }

    _renderItem(data: any) {
        return (
            <Text style={styles.cardListName}>
                {data.index + 1}. {data.item.Country}
            </Text>
        )
    }

    goToViewAllScreen() {
        this.props.navigateTo()
    }

    render() {
        return (
            <View style={styles.cardView}>
                <FlatList
                    data={this.props.data}
                    renderItem={this._renderItem}
                    keyExtractor={(item) => item.CountryCode}
                />
                <Button
                    title={LanguagesStrings.viewAll}
                    color="blue"
                    onPress={() => this.goToViewAllScreen()}
                />
            </View>
        )
    }
}

const width = Dimensions.get('window').width - 20;

const styles = StyleSheet.create({
    cardView: {
        height: 200,
        width: width,
        margin: 5,
        justifyContent: 'center',
        backgroundColor: '#F5FCFF',
        borderRadius: 5,
        shadowColor: 'gray',
        shadowOffset: { width: 0, height: 1 },
        shadowOpacity: 0.8,
        shadowRadius: 2,
        elevation: 5
    },
    cardListName: {
        margin: 5,
        textAlign: 'left',
        fontSize: 17,
        fontFamily: "Avenir-medium"
    }

})

export default TopFiveList;