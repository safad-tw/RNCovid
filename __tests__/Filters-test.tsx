import * as React from "react";
import renderer from "react-test-renderer";
import TopFiveList from "../src/components/countryList/Filters";
import { render, fireEvent } from '@testing-library/react-native';
import '@testing-library/jest-dom'
import Filters from "../src/components/countryList/Filters";

test("Render component", () => {
    const clickFn = jest.fn();
    const buttons = [{label: "recovered"},{label: "deaths"},{abel: "active"}]
    const tree = renderer.create(<Filters data={buttons} onValueChange={clickFn}/>).toJSON();
    expect(tree).not.toBeNull()
  });


  test('Click filter recovered', () => {

    const clickFn = jest.fn();
    const buttons = [{label: "recovered"},{label: "deaths"},{label: "active"}]
    const { getByText } = render(<Filters data={buttons} onValueChange={clickFn}/>)
    fireEvent.press(getByText("recovered"));
  
  });

  test('Click filter deaths', () => {

    const clickFn = jest.fn();
    const buttons = [{label: "recovered"},{label: "deaths"},{label: "active"}]
    const { getByText } = render(<Filters data={buttons} onValueChange={clickFn}/>)
    fireEvent.press(getByText("deaths"));
  
  });

  test('Click filter deaths', () => {

    const clickFn = jest.fn();
    const buttons = [{label: "recovered"},{label: "deaths"},{label: "active"}]
    const { getByText } = render(<Filters data={buttons} onValueChange={clickFn}/>)
    fireEvent.press(getByText("active"));
  
  });


  test('Renders snapshot as expected', () => {
    const clickFn = jest.fn();
    const buttons = [{label: "recovered"},{label: "deaths"},{label: "active"}]
    const tree = render(<Filters data={buttons} onValueChange={clickFn}/>)
    expect(tree).toMatchSnapshot();
  });