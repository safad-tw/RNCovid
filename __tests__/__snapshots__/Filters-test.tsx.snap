// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 20,
        "borderWidth": 2,
        "margin": 5,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
        }
      }
    >
      recovered
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 20,
        "borderWidth": 2,
        "margin": 5,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
        }
      }
    >
      deaths
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 20,
        "borderWidth": 2,
        "margin": 5,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
        }
      }
    >
      active
    </Text>
  </View>
</View>
`;
